src/command/restore/restore.c

```diff
@@ -2068,11 +2069,28 @@ restoreProcessQueue(const Manifest *const manifest, List **const queueList)
         MEM_CONTEXT_END();
 
         // Now put all files into the processing queues
+        const bool isFilterSet = cfgOptionTest(cfgOptFilter);
         for (unsigned int fileIdx = 0; fileIdx < manifestFileTotal(manifest); fileIdx++)
         {
             const ManifestFilePack *const filePack = manifestFilePackGet(manifest, fileIdx);
             const ManifestFile file = manifestFileUnpack(manifest, filePack);
 
+            if (isFilterSet)
+            {
+                Oid dbNode, relNode;
+                Oid spcNode = DEFAULTTABLESPACE_OID;
+                char tsId[32];
+                if (
+                    // If this file is located in the default tablespace
+                    sscanf(strZ(file.name), MANIFEST_TARGET_PGDATA "/" PG_PATH_BASE "/%u/%u", &dbNode, &relNode) == 2 ||
+                    // If this file is located in a non-built-in tablespace
+                    sscanf(strZ(file.name), MANIFEST_TARGET_PGTBLSPC "/%u/%31[^/]/%u/%u", &spcNode, tsId, &dbNode, &relNode) == 4)
```

Comment:
an unused value may be skipped.
```suggestion
                    sscanf(strZ(file.name), MANIFEST_TARGET_PGTBLSPC "/%u/%*31[^/]/%u/%u", &spcNode, &dbNode, &relNode) == 4)
```

Reply:
done